packages/itwin/tree-widget/src/test/trees/VisibilityTreeEventHandler.test.ts

```diff
@@ -115,5 +119,60 @@ describe("VisibilityTreeEventHandler", () => {
       expect(getVisibilityStatus.callCount).to.eq(1);
     });
 
+    it("does not call 'getVisibilityStatus' while changing visibility", async () => {
+      const node1 = createSimpleTreeModelNode("testId1");
+      modelMock.reset();
+      modelMock.setup((x) => x.getNode("testId1")).returns(() => node1);
+      const eventHandler = createHandler({ visibilityHandler });
+      const changes: CheckboxStateChange[] = [{ nodeItem: node1.item, newState: CheckBoxState.On }];
+      const changesSubject = new Subject<CheckboxStateChange[]>();
+      changesSubject.subscribe({
+        next: (change) => {
+          return change;
+        },
+        complete: () => { },
+      });
+
+      changeVisibility.returns(EMPTY);
+      await using(eventHandler, async (_) => {
+        eventHandler.onCheckboxStateChanged({
+          stateChanges: changesSubject,
+        });
+        changesSubject.next(changes);
+        onVisibilityChange.raiseEvent(["testId1"]);
+        changesSubject.complete();
+        onVisibilityChange.raiseEvent(["testId1"]);
+        await flushAsyncOperations();
```

Comment:
Is this needed?

Reply:
I looked at older tests and I think it is better leave this in as raising `onVisibilityChange` event starts an async operation.